501.py

```diff
@@ -0,0 +1,30 @@
+# Definition for a binary tree node.
+# class TreeNode:
+#     def __init__(self, val=0, left=None, right=None):
+#         self.val = val
+#         self.left = left
+#         self.right = right
+class Solution:
+    def findMode(self, root: TreeNode) -> List[int]:
+        def pushleft(stack: List[TreeNode], node: TreeNode):
+            while node:
+                stack.append(node)
+                node = node.left
+
+        modes = []
+        counter = collections.defaultdict(int)
+        stack = []
+        pushleft(stack, root)
+
+        while stack:
+            node = stack.pop()
+            counter[node.val] += 1
+            pushleft(stack, node.right)
+
+        mode_count = max(counter.values())
```

Comment:
The mode of the numbers in this tree can be obtained while you traverse the tree. As soon as you update each `counter[node.val]`, you can compare this latest count with the current maximum count. Therefore, line 24 is not needed.